src/k2/transport/RPCDispatcher.cpp

```diff
@@ -150,43 +157,62 @@ void RPCDispatcher::_handleNewMessage(Request&& request) {
     }
 }
 
-void RPCDispatcher::_send(Verb verb, std::unique_ptr<Payload> payload, TXEndpoint& endpoint, MessageMetadata meta) {
 
+seastar::future<>
+RPCDispatcher::_send(Verb verb, std::unique_ptr<Payload> payload, TXEndpoint& endpoint, MessageMetadata meta) {
+    auto ep = RPC().getServerEndpoint(endpoint.getProtocol());
     auto protoi = _protocols.find(endpoint.getProtocol());
     if (protoi == _protocols.end()) {
         K2WARN("Unsupported protocol: "<< endpoint.getProtocol());
-        return;
+        return seastar::make_ready_future<>();
     }
     auto serverep = protoi->second->getServerEndpoint();
     K2DEBUG("sending message for verb: " << int(verb) << ", to endpoint=" << endpoint.getURL()
-            << ", with server endpoint: " << (serverep ? serverep->getURL() : String("none")));
-    if (serverep && endpoint == *serverep) {
-        // deliver via a future to possibly yield if there is a loop of send/receive requests
-        (void) seastar::sleep(0ns)
-            .then([disp=weak_from_this(), verb, endpoint, meta=std::move(meta), payload=std::move(payload)] () mutable {
-                if (disp) {
-                    // rewind the payload to the correct position
-                    payload->seek(txconstants::MAX_HEADER_SIZE);
-                    meta.setPayloadSize(payload->getDataRemaining());
-                    disp->_handleNewMessage(Request(verb, endpoint, std::move(meta), std::move(payload)));
-                }
-        });
-        return;
+            << ", with server endpoint: " << (serverep ? serverep->getURL() : String("none"))
+            << ", payload size=" << (payload?payload->getSize():0)
+            << ", payload capacity=" << (payload?payload->getCapacity():0));
+    if (_txUseCrossCoreLoopback()) {
+        if (serverep && endpoint == *serverep){
+            payload->seek(txconstants::MAX_HEADER_SIZE);
+            meta.setPayloadSize(payload->getDataRemaining());
+            _handleNewMessage(Request(verb, *RPC().getServerEndpoint(endpoint.getProtocol()), std::move(meta), std::move(payload)));
+            return seastar::make_ready_future<>();
+        }
+        
+        auto core = _url_cores.find(endpoint.getURL());
+        if (core != _url_cores.end()){
+            payload->seek(txconstants::MAX_HEADER_SIZE);
+            meta.setPayloadSize(payload->getDataRemaining());
+            auto return_value = RPCDist().invoke_on(core->second, &k2::RPCDispatcher::_handleNewMessage, Request(verb, *RPC().getServerEndpoint(endpoint.getProtocol()), std::move(meta), std::move(payload)));
```

Comment:
I think you should just (void) this here, with a comment explaining that we don't care about the result of this call since we don't make a promise that we're going to deliver the data.
However, you should chain a handle_exception() continuation so that we can log in case of errors

Reply:
void this invoke_on call and chained a handle_exception.